R/layout.R

```diff
@@ -31,18 +37,23 @@
 #' @inheritParams card
 #' @inheritParams card_body
 #'
-#' @export
 #' @examples
-#'
 #' x <- card("A simple card")
 #' # Always has 2 columns (on non-mobile)
-#' layout_column_wrap(1/2, x, x, x)
-#' # Has three columns when viewport is wider than 750px
-#' layout_column_wrap("250px", x, x, x)
+#' layout_column_wrap(width = 1/2, x, x, x)
+#'
+#' # Automatically lays out three cards into columns
+#' # such that each column is at least 200px wide:
+#' layout_column_wrap(x, x, x)
 #'
+#' # To use larger column widths by default, set `width`.
+#' # This example has 3 columns when the screen is at least 900px wide:
+#' layout_column_wrap(width = "300px", x, x, x)
+#'
+#' @export
 layout_column_wrap <- function(
-  width,
   ...,
+  width = "200px",
```

Comment:
Seems like we have room to explore better default behavior. I haven't fully thought this through, but what if a value of `NA` yielded something like:

```css
grid-template-columns: repeat(auto-fit, minmax(clamp(0px, 200px, 100%), 1fr))
```

This way, if you dump a bunch of elements that are less than 200px wide, you don't get a bunch of empty space (e.g., `layout_column_wrap(!!!replicate(10, actionButton("foo", "Button"), simplify = F))`)

Looking at this also makes me wonder if we should've had `align-content: center;` on the grid from day one 🤔

Reply:
My gut reaction is that `actionButton()` (or shiny inputs in general) should be full width when used as direct children of `layout_column_wrap()`. I recognize that that sort of side-steps the point of your example, but I do think that in many other cases we wouldn't want the clamp behavior.

More broadly, I think we want uniformly even columns from `layout_column_wrap()`, at least when using its public arguments. (Meaning: if you want a non-even column layout you'd define the `grid-template-columns` property via inline styles.) But I think the current proposed default is easier to reason about and closer to what most people want (i.e. put these things in columns but don't let them be smaller than X pixels wide, unless there isn't enough room of course).